fe/src/main/java/org/apache/doris/analysis/FunctionCallExpr.java

```diff
@@ -421,13 +421,23 @@ private void analyzeBuiltinAggFunction(Analyzer analyzer) throws AnalysisExcepti
         }
 
         if (fnName.getFunction().equalsIgnoreCase("percentile_approx")) {
-            if (children.size() != 2) {
-                throw new AnalysisException("percentile_approx(expr, DOUBLE) requires two parameters");
+            if (children.size() != 2 && children.size() != 3) {
+                throw new AnalysisException("percentile_approx(expr, DOUBLE [, B]) requires two or three parameters");
             }
             if (!getChild(1).isConstant()) {
                 throw new AnalysisException("percentile_approx requires second parameter must be a constant : "
                         + this.toSql());
             }
+            if (children.size() == 3) {
+                if (!getChild(2).isConstant()) {
+                    throw new AnalysisException("percentile_approx requires the third parameter must be a constant : "
+                            + this.toSql());
+                }
+                double param_3 = Expr.getConstFromExpr(getChild(2));
```

Comment:
1. Frontend only support portion of all functions, some function can't be compute in Frontend. So better not compute this value, just do this in backend. When input value is not valid, just use default value.
2. param_3 should be renamed to param3

Reply:
Get it